eZ/Publish/Core/FieldType/Validator/StringLengthValidator.php

```diff
@@ -66,6 +74,20 @@ public function validateConstraints($constraints)
             }
         }
 
+        // if no errors above, check if minStringLength is shorter or equal than maxStringLength
+        if (empty($validationErrors) &&
```

Comment:
You could move this awful `if` to a protected method to make it more readable. What about `$this->validateConstraintsOrder($constraints)` ?